### PR TITLE
Fill out the remaining missing docs and configure denying of missing docs from now on

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: src/olm/account/mod.rs [hunk 1]
```diff
@@ -113,17 +115,17 @@ impl Account {
...
     }
 
-    /// Get a reference to the account's public Ed25519 key
+    /// Get a copy of the account's public Ed25519 key
     pub const fn ed25519_key(&self) -> Ed25519PublicKey {
         self.signing_key.public_key()
     }
 
-    /// Get a reference to the account's public Curve25519 key
+    /// Get a copy to the account's public Curve25519 key
```
Comment (andybalaam) [inline src/olm/account/mod.rs:128]: copy to -> copy of

**Review by andybalaam — APPROVED** [review]
Looks good, one tiny change.

File: src/types/mod.rs [hunk 2]
```diff
@@ -35,28 +39,51 @@ impl From<KeyId> for String {
...
     Signature(#[from] SignatureError),
-    /// At least one of the keys did not have contributory behaviour and the
-    /// resulting shared secret would have been insecure.
+
+    /// One or more keys lacked contributory behavior in the Diffie-Hellman
+    /// operation, resulting in an insecure shared secret.
+    ///
+    /// For more details on contributory behavior please refer to the
+    /// [`x25519_dalek::SharedSecret::was_contributory()`] method.
     #[error("At least one of the keys did not have contributory behaviour")]
```
Comment (dkasak) [inline src/types/mod.rs:87]: Should probably change the error to use the same phrasing?

**Review by dkasak — APPROVED** [review]

File: src/olm/account/mod.rs (hunk 1 above)
Comment (poljar) [reply src/olm/account/mod.rs]: 50b8fd3cbeb8bda10bf3cf0d19904fbe7fbc0b5d

File: src/types/mod.rs (hunk 2 above)
Comment (poljar) [reply src/types/mod.rs]: d9e9a70c6ddf3e8bd6b2f6021533806e4ce47aa3

File: src/cipher/mod.rs [hunk 3]
```diff
@@ -83,28 +89,53 @@ pub enum DecryptionError {
...
+    ///
+    /// This key derivation format is typically used for generating individual
+    /// message keys in the Megolm ratchet.
     pub fn new_megolm(&key: &[u8; 128]) -> Self {
         let keys = CipherKeys::new_megolm(&key);
 
         Self { keys }
     }
 
+    /// Creates a new [`Cipher`] from the given 32-byte raw key.
```
[the remark this answers is not in the document]
Comment (poljar) [reply src/cipher/mod.rs:131]: Well it should™ be a 32-byte key, but libolm accepted any length so here we are.

[the remark this answers is not in the document]
Comment (poljar) [reply src/cipher/mod.rs]: 0d528765708ebfee1250a5e175893c619b1ae866

**Pull request merged**